hudi-hadoop-common/src/main/java/org/apache/parquet/avro/HoodieAvroReadSupport.java

```diff
@@ -46,11 +46,7 @@ public HoodieAvroReadSupport() {
   @Override
   public ReadContext init(Configuration configuration, Map<String, String> keyValueMetaData, MessageType fileSchema) {
     boolean legacyMode = checkLegacyMode(fileSchema.getFields());
-    // support non-legacy list
-    if (!legacyMode && configuration.get(AvroWriteSupport.WRITE_OLD_LIST_STRUCTURE) == null) {
```

Comment:
This line has been extracted and now I'm forcefully applying configuration even if something is set by the user, since if the user has `AvroWriteSupport.WRITE_OLD_LIST_STRUCTURE=true`, then the case with rewriting 3 levels into 2 levels will either fail/silent data loss - basically scenarios 5 and 6 from the test suite.
This is opposite use case as we faced initially.

Reply:
Essentially you can reproduce it if run with the old code - scenarios 5 and 6 in the test.